.github/workflows/check.yaml

```diff
@@ -0,0 +1,44 @@
+name: Check workflow running linter, unit and functional tests
+
+on:
+  workflow_call:
+  workflow_dispatch:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches: [ master, main ]
+    paths-ignore:
+      - '**.md'
+      - '**.rst'
+
+concurrency:
+      # We do not want to run multiple jobs for single PR.
+      group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+      cancel-in-progress: true
+
+jobs:
+  lint-unit:
+    name: Lint checkers and Unit tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install tox
+    - name: Run lint checkers
+      run: tox -e lint
+    - name: Run unit tests
+      run: tox -e unit
+    - name: Save PR number to file
+      run: echo ${{ github.event.number }} > PR_NUMBER.txt
+    - name: Archive PR number
+      uses: actions/upload-artifact@v4
+      with:
+        name: PR_NUMBER
+        path: PR_NUMBER.txt
```

Comment:
I'm curious about why we save the PR number?

Reply:
it could be a leftover from me removing the coverage checks. It doesn't look like it's needed anymore

Reply:
found it, I think it's a leftover from sonar integration. See for example https://github.com/canonical/prometheus-juju-exporter/blob/65c5d436a7ce45f9824565b0144d939d1a0c2f36/.github/workflows/sonar.yaml